apps/tx-builder/src/components/forms/SolidityForm.tsx

```diff
@@ -0,0 +1,202 @@
+import { useState } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { ButtonLink, Switch, Text } from '@gnosis.pm/safe-react-components';
+import styled from 'styled-components';
+
+import { ContractInterface } from '../../hooks/useServices/interfaceRepository';
+import {
+  ADDRESS_FIELD_TYPE,
+  AMOUNT_FIELD_TYPE,
+  CONTRACT_METHOD_FIELD_TYPE,
+  HEX_ENCODED_DATA_FIELD_TYPE,
+} from './fields/fields';
+import Field from './fields/Field';
+import { Examples } from '../Examples';
+import { encodeToHexData } from '../../utils';
+
+function SolidityForm({
+  id,
+  onSubmit,
+  getAddressFromDomain,
+  initialValues,
+  nativeCurrencySymbol,
+  networkPrefix,
+  contract,
+  children,
+}: SolidityFormPropsTypes) {
+  const [showExamples, setShowExamples] = useState<boolean>(false);
+  const [showHexEncodedData, setShowHexEncodedData] = useState<boolean>(false);
+
+  const { handleSubmit, control, setValue, watch, getValues, reset, clearErrors } = useForm<SolodityFormValues>({
+    defaultValues: initialValues,
+    mode: 'onTouched', // This option allows you to configure the validation strategy before the user submits the form
+  });
+
+  const contractMethodIndex = watch(CONTRACT_METHOD_INDEX_FIELD_NAME);
+  const contractMethod = contract?.methods[Number(contractMethodIndex)];
+  const contractFields = contractMethod?.inputs || [];
+  const showContractFields = !!contract && !showHexEncodedData;
+  const isPayableMethod = !!contract && contractMethod?.payable;
+
+  const isValueInputVisible = showHexEncodedData || !contract || isPayableMethod;
+
+  function onClickShowHexEncodedData(checked: boolean) {
+    const contractFieldsValues = getValues(CONTRACT_VALUES_FIELD_NAME);
+
+    if (checked && contractMethod) {
+      const encodeData = encodeToHexData(contractMethod, contractFieldsValues);
+      setValue(HEX_ENCODED_DATA_FIELD_NAME, encodeData || '');
+    }
+    setShowHexEncodedData(checked);
+  }
+
+  function submitAndResetForm(values: SolodityFormValues) {
+    onSubmit(values);
+    reset(initialValues);
+    setTimeout(clearErrors, 0);
+  }
+
+  return (
+    <form id={id} onSubmit={handleSubmit(submitAndResetForm)} noValidate>
+      {/* To Address field */}
+      <Field
```

Comment:
A lot of these props are used across multiple fields, e.g. `fullWidth`. Maybe you could export them into a reuseable component?